plugins/FileUpload/class.fileupload.plugin.php

```diff
@@ -827,7 +827,11 @@ public function PostController_Upload_Create($Sender) {
          $this->EventArguments['OriginalFilename'] = $FileName;
          $Handled = FALSE;
          $this->EventArguments['Handled'] =& $Handled;
-         $this->FireAs('Gdn_Upload')->FireEvent('SaveAs');
+         if($ImageType !== FALSE) {
+            $this->FireAs('Gdn_Upload')->FireEvent('SaveImageAs');
```

Comment:
What was the purpose behind throwing a differently-named event here? Typically we'd want to do something like add an event argument for `$ImageType` rather than throw two different events.